File: II-semester/Lista 3/Figury: C++/src/Figury.cpp
#include "Kolo.hpp"
#include "Pieciokat.hpp"
#include "Szesciokat.hpp"
#include "Kwadrat.hpp"
#include "Prostokat.hpp"
#include "Romb.hpp"
#include <math.h>

Figura::Figura(){}
Figura::~Figura(){}
Czworokat::Czworokat(){}
Czworokat::~Czworokat(){}

Kolo::Kolo(double r)
{
    promien = r;
}

double Kolo::obwod()
{
    return 2.0 * promien * M_PI;
}

double Kolo::pole()
{
    return M_PI * pow(promien, 2);
}

const char* Kolo::wezNazwa()
{
    return "Koło";
}

Kolo::~Kolo(){}

Pieciokat::Pieciokat(double a)
{
    bok = a;
}

double Pieciokat::obwod()
{
    return 5.0 * bok;
}

double Pieciokat::pole()
{
    return 1.0/4.0 * pow(bok, 2) * sqrt(25.0 + 10.0 * sqrt(5.0));
}

const char* Pieciokat::wezNazwa()
{
    return "Pięciokąt";
}

Pieciokat::~Pieciokat(){}

Szesciokat::Szesciokat(double a)
{
    bok = a;
}

double Szesciokat::obwod()
{
    return 6.0 * bok;
}

double Szesciokat::pole()
{
    return 3.0/2.0 * pow(bok, 2) * sqrt(3.0);
}

const char* Szesciokat::wezNazwa()
{
    return "Sześciokąt";
}

Szesciokat::~Szesciokat(){}

Kwadrat::Kwadrat(double a)
{
    bok1 = a;
}

double Kwadrat::obwod()
{
    return 4.0 * bok1;
}

double Kwadrat::pole()
{
    return pow(bok1, 2);
}

const char* Kwadrat::wezNazwa()
{
    return "Kwadrat";
}

Kwadrat::~Kwadrat(){}

Prostokat::Prostokat(double a, double b)
{
    bok1 = a;
    bok2 = b;
}

double Prostokat::obwod()
{
    return 2.0 * bok1 + 2.0 * bok2;
}

double Prostokat::pole()
{
    return bok1 * bok2;
}

const char* Prostokat::wezNazwa()
{
    return "Prostokąt";
}

Prostokat::~Prostokat(){}

Romb::Romb(double a, double b)
{
    bok1 = a;
    kat = b;
}

double Romb::obwod()
{
    return 4.0 * bok1;
}

double Romb::pole()
{
    return pow(bok1, 2) * sin(kat * (M_PI/180.0));
}

const char* Romb::wezNazwa()
{
    return "Romb";
}

Romb::~Romb(){}

File: II-semester/Lista 3/Figury: Java/Figury.java
import static java.lang.Math.*;
import java.util.*;

interface Pole
{
    public double pole();
}

interface Obwod
{
    public double obwod();
}

interface WezNazwa
{
    public String wezNazwa();
}

abstract class Figura implements Pole, Obwod, WezNazwa
{
    
}

abstract class Czworokąt extends Figura
{

}

class Romb extends Czworokąt
{
    private double bok;
    private double kąt;
    private String nazwa = "Romb";
    
    Romb(ArrayList <Double> argumenty)
    {
        bok = argumenty.get(0);
        kąt = argumenty.get(4);
    }
    public double pole()
    {
        return pow(bok, 2) * sin(toRadians(kąt));
    }
    public double obwod()
    {
        return 4.0 * bok;
    }
    public String wezNazwa()
    {
        return nazwa;
    }
}

class Prostokąt extends Czworokąt
{
    private double bok1;
    private double bok2;
    private String nazwa = "Prostokąt";
    
    Prostokąt(ArrayList <Double> argumenty)
    {
        argumenty.remove(90.0);
        bok1 = argumenty.get(0);
        bok2 = argumenty.get(2);
    }
    
    public double pole()
    {
        return bok1 * bok2;
    }
    public double obwod()
    {
        return 2.0 * bok1 + 2.0 * bok2;
    }
    public String wezNazwa()
    {
        return nazwa;
    }
}

class Kwadrat extends Czworokąt
{
    private double bok;
    private String nazwa = "Kwadrat";
    
    Kwadrat(ArrayList <Double> argumenty)
    {
        argumenty.remove(90.0);
        bok = argumenty.get(0);
    }
    
    public double pole()
    {
        return bok * bok;
    }
    public double obwod()
    {
        return 4.0 * bok;
    }
    public String wezNazwa()
    {
        return nazwa;
    }
}

class Koło extends Figura
{
    private double promień;
    private String nazwa = "Koło";
    
    Koło(double r)
    {
        promień = r;
    }
    public double pole()
    {
        return PI * pow(promień, 2);
    }
    public double obwod()
    {
        return 2.0 * PI * promień;
    }
    public String wezNazwa()
    {
        return nazwa;
    }
}

class Pięciokąt extends Figura
{
    private double bok;
    private String nazwa = "Pięciokąt";
    
    Pięciokąt(double a)
    {
        bok = a;
    }
    public double pole()
    {
        return 1.0/4.0 * pow(bok, 2) * sqrt(25.0 + 10.0 * sqrt(5.0));
    }
    public double obwod()
    {
        return 5.0 * bok;
    }
    public String wezNazwa()
    {
        return nazwa;
    }
}

class Sześciokąt extends Figura
{
    private double bok;
    private String nazwa = "Sześciokąt";
    
    Sześciokąt (double a)
    {
        bok = a;
    }
    public double pole()
    {
        return 3.0/2.0 * pow(bok, 2) * sqrt(3.0);
    }
    public double obwod()
    {
        return 6.0 * bok;
    }
    public String wezNazwa()
    {
        return nazwa;
    }
}
